stdlib/public/core/Filter.swift

```diff
@@ -98,40 +98,11 @@ public struct LazyFilterSequence<Base : Sequence>
 /// - Note: The performance of advancing a `LazyFilterIndex`
 ///   depends on how sparsely the filtering predicate is satisfied,
 ///   and may not offer the usual performance given by models of
-///   `ForwardIndex`.
-public struct LazyFilterIndex<
-  BaseElements : Collection
-> {
-  /// Returns the next consecutive value after `self`.
-  ///
-  /// - Precondition: The next value is representable.
-  ///
-  /// - Complexity: Amortized O(M), where M is the average distance in
-  ///   the base collection between elements that satisfy the
-  ///   predicate.
-  ///
-  /// - Note: This operation may not satisfy the expected complexity
-  ///   for models of `ForwardIndex`.
-  public func successor() -> LazyFilterIndex {
-    for p in base.successor()..<_baseElements.endIndex {
-      if _include(_baseElements[p]) {
-        return LazyFilterIndex(
-          _baseElements: _baseElements, base: p, _include: _include)
-      }
-    }
-    return LazyFilterIndex(
-      _baseElements: _baseElements, base: _baseElements.endIndex,
-      _include: _include)
-  }
-
-  internal let _baseElements: BaseElements
+///   `Collection`.
+public struct LazyFilterIndex<Base : Collection> : Comparable {
```

Comment:
I'd say that it makes sense, at least as a way of preventing mistakes of using base indices with the filtered collection.